### ISSUE TITLE
MPAndroidChart Making Curved Graphs

### ISSUE THREAD

**Issue by Edonfreiner** [PhilJay/MPAndroidChart#3258]
I am using the MPAndroidChart to make a graph of the sun movements and marking different points during the day. At this moment it looks like this: [![enter image description here][1]][1]
  

When I change it to `dataSet.setMode(LineDataSet.Mode.CUBIC_BEZIER);`
It doesnt make it look curved just weird: 

[![enter image description here][2]][2]

A really small change at the top. And when I do `dataSet.setMode(LineDataSet.Mode.HORIZONTAL_BEZIER);`

It also has this weird loopy effect. 

What am I doing wrong, are the points on my graph wrong?

I would like to get this type of effect:
[![enter image description here][3]][3]

Please help, thanks

  [1]: https://i.stack.imgur.com/MJiWt.png
  [2]: https://i.stack.imgur.com/hIS0N.png
  [3]: https://i.stack.imgur.com/9ZyKs.png

**Comment by JStav** [comment]
Try lowering the cubic intensity with ```dataSet.setCubicIntensity()```. Default is 0.2f. Mine is set to 0.1f.

**Comment by Edonfreiner** [comment]
Hey @JStav Thanks for replying, this didnt work. The curve works with only 3 points so lets say, (5,5) (0,0) (-5,5) but when I add more points this stops working and looks very linear.

**Here are my points:**

alos:                                                                     X: 240.0 ,Y: 508.0
zmanTallisAndTefillin:                                        X: 263.0, Y: 462.0
netzHaChama:                                                    X: 302.0, Y: 386.0
sofZmanShemahMagenAvrah:                          X: 368.0, Y: 254.0
sofZmanShemahGra:                                         X: 398.0, Y: 193.0
sofZmanTefilla:                                                  X: 430.0, Y: 129.0
chatzos:                                                              X: 495.0, Y: 0.0
minchaGedola:                                                   X: 511.0, Y: 31.0
plagHaMincha:                                                   X: 647.0, Y: 305.0
shkiah:                                                                X: 687.0, Y: 384.0
tzesHakochavim:                                               X: 719.0, Y: 448.0



Thanks again @JStav

**Comment by JStav** [comment]
Yeah, smoothing leaves a lot to be desired in this library. We only really have CUBIC_BEZIER and HORIZONTAL_BEZIER. I have this problem too and there's not much you can really do. 

Other libraries like D3 for js have many more options that work better. Unfortunate we do not have that. It's probably a good candidate for a feature addition/pull request.

**Comment by Edonfreiner** [comment]
Thanks @JStav.

Is there anyway to to get this to work?

Maybe make the 3 points and plot the rest of the them on that line but on a different lineset?

Or a different Library maybe?

Can I convert the D3 library and use it with android? Is that possible?
Thanks, I would really need this to work.

All The best

**Comment by JStav** [comment]
If it works every 3 points then yeah it would be a workaround to just make a new line data set every 3 points. You would just have to make sure that it looks good in all possible scenarios.

D3 as far as I know is a JavaScript only library so I don't think you can use that unless you plop it all into a WebView but that's kind of messy.

**Comment by Edonfreiner** [comment]
Ok, so I made a chart for the first 3 points:

![screen shot 2017-07-18 at 3 05 01 pm](https://user-images.githubusercontent.com/19429507/28334924-aa19fbda-6bca-11e7-8591-35eb08ac5a73.png)

And your saying that I should plot the rest on a new line with out any color?
That should work?

How do I plot multiple linesets with one chart?

**Comment by JStav** [comment]
You can use a ScatterDataSet and use CombinedData instead of LineData. 

Is this graph supposed to have different sets of data or will it always be the same? If the data changes you probably can't use this method because there's no way to tell which points you need to separate to make the curve work other than trial and error.

**Comment by Edonfreiner** [comment]
What I am trying to do is get a sun path. As shown below:

![sunpath](https://user-images.githubusercontent.com/19429507/28335558-ce3fb822-6bcc-11e7-9d55-80bb590ed68e.gif)

The multiple lines in this graph represents different months of the year.

I have specific points of the day according to the sun and the times they occur. For example sun rise, sun set and midday (half way between sunrise and sunset) along with other times.

I thought I can plot them on a graph dividing the time by 24 to get the X and the time by midday (around 12) to get the Y. and then plot them on a graph.

These times change every day, however they are all relative to one another.

My end goal is to hopefully have something like this: 

![time-travel-changing-solar-view-apple-watch-screenshot](https://user-images.githubusercontent.com/19429507/28335720-631fe7c8-6bcd-11e7-8b3c-dd53667b0b81.jpg)

Where each of those images are separate days.


Do you think this is at all possible? 
Am I in way over my head?

Thanks.

**Comment by JStav** [comment]
I think you will run into many other problems with the library on the way, even if you fix this one. Unfortunately I don't really know a good alternative or something that could help you achieve this so I can't be of much help with that, sorry. Might have to roll your own custom solution.

**Comment by Edonfreiner** [comment]
hey, Thank you for your help I figured it out. I used the elevation of sun to graph the line and then plotted the points in time on the a different data set, on that graph.

I was wondering if it is possible to set highlight effect for only a specific data set. thanks
